tests/core/test_regions.py

```diff
@@ -200,26 +200,262 @@ def test_polygon_sampling():
+        "tools" / "meshes" / "classic_plane.obj.bz2",
+        dimensions=(20,20,10), rotation=(math.radians(-90), 0, math.radians(-10)))
+    MeshSurfaceRegion.fromFile(Path(".").parent.parent.parent /
+        "tools" / "meshes" / "classic_plane.obj.bz2", dimensions=(20,20,10),
+        rotation=(math.radians(-90), 0, math.radians(-10)))
+
+def test_mesh_operation_blender():
+    r1 = BoxRegion(position=(0,0,0), dimensions=(1,1,1), engine="blender")
+    r2 = BoxRegion(position=(0,0,0), dimensions=(2,2,2), engine="blender")
+
+    r = r1.intersect(r2)
+
+def test_mesh_operation_scad():
+    r1 = BoxRegion(position=(0,0,0), dimensions=(1,1,1), engine="scad")
+    r2 = BoxRegion(position=(0,0,0), dimensions=(2,2,2), engine="scad")
+
+    r = r1.intersect(r2)
+
+def test_mesh_volume_region_sampling():
+    r = BoxRegion(position=(0,0,0), dimensions=(2,2,2))
+    pts = [r.uniformPointInner() for _ in range(100)]
+
+    for x, y, z in pts:
+        assert -1 <= x <= 1
+        assert -1 <= y <= 1
+        assert -1 <= z <= 1
+
+def test_mesh_surface_region_sampling():
+    r = BoxRegion(position=(0,0,0), dimensions=(2,2,2)).getSurfaceRegion()
+    pts = [r.uniformPointInner() for _ in range(100)]
+
+    for x, y, z in pts:
+        assert  x == 1 or x == -1 or \
+                y == 1 or y == -1 or \
+                z == 1 or z == -1
+
+def test_path_region():
+    points_2d = [(0,0), [0,1], (1,1), (1,0)]
+    points_3d = [(3,3,0), (3,4,1), [4,4,2], (4,3,3)]
+    points_mixed = [(6,6), (6,7,1), (7,7,2), [7,6,3]]
+    polylines = [points_2d, points_3d, points_mixed]
+
+    # Test creation
+    r1 = PathRegion(points=points_2d)
+    r2 = PathRegion(points=points_3d)
+    r3 = PathRegion(points=points_mixed)
+    r4 = PathRegion(polylines=polylines)
+
+    with pytest.raises(TypeError):
+        PathRegion(points=polylines)
+
+    with pytest.raises(TypeError):
+        PathRegion(polylines=points_mixed)
+
+    # Test simple containment
+    assert r1.containsPoint((0,0))
+    assert r1.containsPoint((0,1))
+    assert r1.containsPoint((0,0.5))
+    assert not r1.containsPoint((0,0,1))
+    assert not r1.containsPoint((0,-1))
+
+    assert r2.containsPoint((3,3,0))
+    assert r2.containsPoint((3,4,1))
+    assert r2.containsPoint((3,3.5,0.5))
+    assert not r2.containsPoint((3,3,1))
+    assert not r2.containsPoint((-2,-2,-2))
+
+    # Test sampling and containment of samples
+    for _ in range(100):
+        target_region = random.choice([r1,r2,r3,r4])
+        sampled_pt = target_region.uniformPointInner()
+        assert target_region.containsPoint(sampled_pt)
+
+    # Test properties
+    assert r1.AABB == ((0,1),(0,1),(0,0))
+    assert r2.AABB == ((3,4),(3,4),(0,3))
+    assert r3.AABB == ((6,7),(6,7),(0,3))
+    assert r4.AABB == ((0,7),(0,7),(0,3))
+
+def test_mesh_polygon_intersection():
+    r1 = BoxRegion(position=(0,0,0), dimensions=(3,3,2))
+    r2 = CircularRegion((0,0), 1, resolution=64)
```

Comment:
`r2` is contained in `r1`; this test would be more interesting if the intersection actually restricted `r2` (beyond making it 2D).

Reply:
Good call, I made the change.